### PR TITLE
A couple small fuzz tests and regression test corpus.

### PR DESCRIPTION
**Pull request by rphmeier**
This introduces 2 fuzz-tests, one for the more general `round` module, and one for the `vote_graph` module. Both operate over a hardcoded chain with 3 forks, where each block can be uniquely identified by a 4-bit number. This also replaces most of the hashmaps with BTreeMaps, for two reasons:
  - HashMaps can be attacked in a `no-std` environment where the hash function is predictable (e.g. blockchain runtimes)
  - I didn't want the fuzzer to explore any PRNG - all randomness in the fuzz-tests comes from the input.

AFL and Libfuzzer are supported.

### CODE REVIEW

File: src/fuzz_helpers.rs [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+	for i in 0..N {
+		let target_hash = match stream.read_nibble() {
+			None => return,
+			Some(h) => h,
+		};
+
+		let target_hash = match FuzzChain.ancestry(prevote_ghost.0, target_hash) {
+			Ok(_) => target_hash,
+			Err(_) => prevote_ghost.0,
+		};
```
[the remark this answers is not in the document]
Comment (rphmeier) [reply src/fuzz_helpers.rs]: The `prevote_ghost` here is "our" perceived prevote-ghost, not that of the voter we are simulating. We limit the set of simulated prevote-ghosts to those that are equal to or descendants of our prevote-ghost

File: src/fuzz_helpers.rs [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+		graph.insert(target_hash, target_number, Vote(0, 1), &FuzzChain).unwrap();
+
+		let new_prevote_ghost = graph.find_ghost(Some(prevote_ghost.clone()), |v| v.0 >= T).unwrap();
+		assert_eq!(new_prevote_ghost, prevote_ghost, "should not change");
+
+		if i < T { return }
+
+		let remaining = N - i - 1;
+		let possible_equivocations = 33;
+		let possible_to_precommit = |v: &Vote| v.1 + remaining + possible_equivocations >= T;
```
[the remark this answers is not in the document]
Comment (rphmeier) [reply src/fuzz_helpers.rs]: this calculation is slightly faster which might be amenable to fuzzing. I agree it's not perfectly accurate

File: src/fuzz_helpers.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (romanb) [reply src/fuzz_helpers.rs]: I see, thanks!

**Comment by rphmeier** [comment]
@romanb Can you make a follow-up PR with all suggested amendments?

**Comment by romanb** [comment]
> @romanb Can you make a follow-up PR with all suggested amendments?

Sure, will do.

**Pull request merged**